python/cudf/cudf/utils/_numba.py

```diff
@@ -135,7 +132,9 @@ def _setup_numba():
             if driver_version < (12, 0):
```

Comment:
Can we update the comment above to mention pynvjitlink and the corresponding role of that package? This comment:
```
    # ptxcompiler is a requirement for cuda 11.x packages but not
    # cuda 12.x packages. However its version checking machinery
    # is still necessary. If a user happens to have ptxcompiler
    # in a cuda 12 environment, it's use for the purposes of
    # checking the driver and runtime versions is harmless
```

Reply:
@brandon-b-miller I would generally advocate reviewing this entire file and any other files that relate to ptxcompiler/pynvjitlink to make sure things are named sensibly, etc. in a way that will support both CUDA 11 and CUDA 12+. I want the code comments and docs to reflect the implemented design going forward.

Keep in mind that we don't want to name things "CUDA 12" in the code if we can avoid it if it is likely that later versions will act in the same way.

Reply:
how about something like https://github.com/rapidsai/cudf/pull/14763/commits/7dbf9f22dbb63b6d33cfd9ad783dee60ec41555c ?

Reply:
> In a CUDA 12.x environment, ptxcompiler provides version checking, but not MVC directly

Is this true? We don't use ptxcompiler in CUDA 12 environments. No environment should have both ptxcompiler and pynvjitlink installed at the same time.

Reply:
It's technically `_ptxcompiler.py` in this case - our slimmed down, vendored version of the few functions we need.

Reply:
Ooooo. But I don't know how to distinguish `ptxcompiler` the package (only used when on CUDA 11) from `_ptxcompiler.py` the internal helper file (always active) from the text of this comment. Documenting that kind of thing clearly is what I want to achieve before merging this.

Reply:
some reworking in https://github.com/rapidsai/cudf/pull/14763/commits/e8a90b9ee26c8e365f5df8e705ed7e8c9f21681a

Reply:
Much clearer! Thanks for iterating on this.